### PR TITLE
Add Page method "insert_htmlbox"

### PR DESCRIPTION
**Pull request by JorjMcKie**
Note:
I removed previous versions because of other fundamental changes.

### CODE REVIEW

**Comment by JorjMcKie** [comment]
Made requested changes.
Decided to after all go away from `scale=True` to a float scale_low, which accepts values inside interval [0,1].
1 means do not scale, 0 is unlimited scale, values in between correspond to scale_max values in `Story.fit_scale()`.

File: src/utils.py [hunk 1]
```diff
@@ -1943,23 +1946,25 @@ def insert_htmlbox(
...
     # ----------------------------------------------------------------
-    if scale is True:
-        scale_max = None
-    else:
+    if scale_low == 1:
         scale_max = 1
+    elif scale_low > 0:
+        scale_max = 1 / scale_low
+    else:
+        scale_max = None
```
Comment (julian-smith-artifex-com) [inline src/utils.py:1954]: Can this be `scale_max = None if scale_low == 0 else 1 / scale_low` ?

File: src/utils.py [hunk 2]
```diff
@@ -1943,23 +1946,25 @@ def insert_htmlbox(
...
 
     fit = story.fit_scale(temp_rect, scale_min=1, scale_max=scale_max)
 
     if fit.big_enough is False:  # there was no fit
-        return (-1, 0)
+        return (-1, scale_low)
```
Comment (julian-smith-artifex-com) [inline src/utils.py:1959]: Not 100% sure, but i thought the docs said we would return (-1, 0) in this case?

File: src/utils.py (hunk 1 above)
Comment (JorjMcKie) [reply src/utils.py:1954]: sure

File: src/utils.py (hunk 2 above)
Comment (JorjMcKie) [reply src/utils.py:1959]: I changed this accordingly

**Comment by julian-smith-artifex-com** [comment]
test_quick is failing because tests do `rc, factor = page.insert_htmlbox(rect, text, rotate=rot, scale=False)`, and  i think `scale=False` now needs to be `scale_low=1`.

**Pull request merged**